src/control/destination/resolution.rs

```diff
@@ -175,7 +190,21 @@ where
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
         loop {
             match self.query.poll() {
-                Ok(Async::NotReady) => return Ok(Async::NotReady),
+                Ok(Async::NotReady) => {
+                    match self.updater.hangup.poll() {
```

Comment:
Why is this only called in the unready state? Shouldn't this be called before polling the underlying client?